File: cs1010/miscellaneous-exercises/practice-s01p02/FreezerV2.java
import java.lang.Math;
import java.text.DecimalFormat;
import java.util.Scanner;

class FreezerV2 {

    private boolean isTest = false;
    
    // Computes value of small t
    // Precon: User has not entered values of hours and minutes
    // Postcon: Compute temperature
    private double computesSmallT() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter hours and minutes since power failure: ");
        double hours = sc.nextDouble();
        double minutes = sc.nextDouble();
        sc.close();

        minutes /= 60; // Convert minutes in terms of hours
        if (isTest) {
            System.out.println("hours = " + hours + ", minutes = " + minutes);
        }
        return (hours + minutes);
    }

    // Computes temperature
    // Precon: smallT has been calculated
    // Postcon: Displays temperature
    private double computesTemperature(double smallT) {
        double numerator = (4.0 * Math.pow(smallT, 10.0));
        double denominator = (2.0 + Math.pow(smallT, 9.0));

        return ((numerator / denominator) - 20.0);
    }

    // Displays temperature
    // Precon: temperature has been calculated
    // Postcon: Nil
    private void displaysTemperature(double temperature) {
        DecimalFormat df = new DecimalFormat("0.00");
        System.out.println("Temperature in freezer = " + df.format(temperature));
    }

    private void run() {
        double smallT = computesSmallT();
        double temperature = computesTemperature(smallT);
        displaysTemperature(temperature);
    }
    
    public static void main(String[] args) {
        FreezerV2 obj = new FreezerV2();
        obj.run();
    }
}

File: modules/programming-methodology/miscellaneous-exercises/practice-s01p02/freezerV2.c
/*
 *  freezerV2.c
 *  Unit 4 Exercise 6: Estimate temperature in a freezer
 *  given the elapsed time since power failure.
 *  New formula: T = (4*t^10 / (t^9+2)) - 20
 *  SHEIKH UMAR
 */
#include <stdio.h>
#include <math.h>

int main(void) {
	int hours, minutes;
	float temperature;  // Temperature in freezer

	// Get the hours and minutes
	printf("Enter hours and minutes since power failure: ");
	scanf("%d %d", &hours, &minutes);

	float small_t = (float) hours + (((float) minutes) / 60.0);

	temperature = (4.0 * pow(small_t, 10.0)) / (2.0 + pow(small_t, 9.0)) - 20.0;

	printf("Temperature in freezer = %.2f\n", temperature);

	return 0;
}
